File: BIT.cpp
class BIT {
private:
    int n;
    vector<int> a;

public:
    BIT(int n) : n(n + 1), a(n + 1, 0) {}

    void update(int i, int val) {
        while (i < n) {
            a[i] += val;
            i += (i & (-i));
        }
    }

    int sum(int i) {
        int s = 0;
        while (i > 0) {
            s += a[i];
            i -= (i & (-i));
        }
        return s;
    }
};

File: BIT.py
class BIT:
    def __init__(self, n):
        self.n = n + 1
        self.a = [0] * self.n

    def update(self, i, val):
        while i < self.n:
            self.a[i] += val
            i += (i & (-i))

    def sum(self, i):
        s = 0
        while i > 0:
            s += self.a[i]
            i -= (i & (-i))
        return s
